modules/custom/az_news/az_news_export/src/Plugin/views/row/AZNewsDataFieldRow.php

```diff
@@ -56,152 +83,191 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function render($row) {
     $output = [];
-    // Provider a helper for image serialization.
-    $image_serializer = function ($value, $entity) {
-      $item = [];
-      if (!empty($value)) {
-        $media = $this->entityTypeManager->getStorage('media')->load($value);
-        if (!empty($media) && $media->access('view') && $media->hasField('field_media_az_image')) {
-          if (!empty($media->field_media_az_image->entity)) {
-            /** @var \Drupal\file\FileInterface $image */
-            $image = $media->field_media_az_image->entity;
-            $item['fid'] = $image->id();
-            $item['uuid'] = $image->uuid();
-            $item['original'] = $image->createFileUrl(FALSE);
-            $uri = $image->getFileUri();
-            $styles = [
-              'thumbnail' => 'az_enterprise_thumbnail',
-              'thumbnail_small' => 'az_enterprise_thumbnail_small',
-            ];
-            foreach ($styles as $key => $style_id) {
-              $image_style = ImageStyle::load($style_id);
-              if (!empty($image_style)) {
-                $item[$key] = $image_style->buildUrl($uri);
-              }
-            }
-            if (!empty($media->field_media_az_image->alt)) {
-              $item['alt'] = $media->field_media_az_image->alt;
-            }
-          }
+    $entity = $row->_entity;
+    $entity_type = $entity->getEntityTypeId();
+    $entity_bundle = $entity->bundle();
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions($entity_type, $entity_bundle);
+    $serializable_entity_types = self::$serializableEntityTypes;
+    foreach ($this->view->field as $field_name => $field) {
+      $field_definition = $field_definitions[$field_name];
+      $value = !empty($this->rawOutputOptions[$field_name]) ? $field->getValue($row) : $field->advancedRender($row);
+      if (empty($value) || empty($field_definition)) {
+        continue;
+      }
+      foreach ($serializable_entity_types as $target_type) {
+        if ($this->isReferenceFieldOfType($field_definition, $target_type)) {
```

Comment:
Here is where i check whether the field targets or allows the entity type in question to be referenced.